ring/func/base.py

```diff
@@ -749,7 +751,11 @@ class BaseStorage(object):
 
     def __init__(self, rope, backend):
         self.rope = rope
-        self.backend = backend
+        self._backend = backend
+
+    @property
+    def backend(self):
+        return self._backend.get() if isinstance(self._backend, contextvars.ContextVar) else self._backend
```

Comment:
It seems `contextvars.ContextVar` raises attribute error when `contextvars is None`